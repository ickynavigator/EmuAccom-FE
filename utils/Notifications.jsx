import { showNotification } from "@mantine/notifications";
import React from "react";
import { Check, X } from "tabler-icons-react";

export const ICON_STROKE_WIDTH = 1;

export const notificationBase = props => {
  showNotification({
    id: props.id,
    disallowClose: props.disallowClose,
    onClose: e => (props.closeCB !== undefined ? props.closeCB(e) : () => {}),
    onOpen: e => (props.openCB !== undefined ? props.openCB(e) : () => {}),
    autoClose: props.autoClose,
    title: props.title,
    message: props.message,
    color: props.color,
    icon: props.icon,
    className: props.className,
    style: props.style,
    sx: props.sx,
    loading: props.loading,
  });
};

export const Notifications = {
  success: (message, options) => {
    notificationBase({
      ...options,
      message,
      color: "green",
      icon: <Check strokeWidth={ICON_STROKE_WIDTH} color="green" />,
    });
  },
  error: (message, options) => {
    notificationBase({
      ...options,
      message,
      color: "red",
      icon: <X strokeWidth={ICON_STROKE_WIDTH} color="red" />,
    });
  },
};

export default Notifications;
