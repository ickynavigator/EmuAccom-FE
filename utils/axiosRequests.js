import axios from "axios";

/** REQUEST HELPERS */
export const serverBase = (environment = "development") => {
  switch (environment) {
    case "production":
      return "https://emuaccom-be-production.up.railway.app";
    case "development":
    default:
      return process.env.NEXT_PUBLIC_SERVER_URL;
  }
};
export const serverURL = `${serverBase(process.env.NODE_ENV)}/v1/api`;
export const axiosResolvers = {
  "400-401": error => {
    if (error.response.status === 401) {
      return Promise.resolve(error);
    }
    if (error.response.status === 400) {
      return Promise.resolve(error);
    }
    return Promise.reject(error);
  },
};
export const paramParser = params => {
  let paramString = "?";
  for (let i = 0; i < params.length; i += 1) {
    if (params[i].val.length > 1)
      paramString += `${params[i].key}=${params[i].val}&`;
  }

  return paramString;
};

/** REQUESTS */
export const signInRequest = data => {
  // AXIOS INTERCEPTOR
  axios.interceptors.response.use(
    response => response,
    error => {
      if (error.response.status === 401) {
        return Promise.resolve(error);
      }
      return Promise.reject(error);
    },
  );

  const postData = {
    email: data.email,
    password: data.password,
  };
  return axios.post(`${serverURL}/users/login`, postData);
};

export const signUpRequest = data => {
  // AXIOS INTERCEPTOR
  axios.interceptors.response.use(
    response => response,
    axiosResolvers["400-401"],
  );
  const postData = {
    email: data.email,
    password: data.password,
    firstName: data.fName,
    lastName: data.fName,
    type: data.type,
  };
  return axios.post(`${serverURL}/users`, postData);
};

export const fetchDorms = ({
  searchParam = "",
  search = "",
  pageSize = 10,
  pageNumber = 1,
  paginate = true,
}) => {
  const parsedParams = paramParser([
    { key: "param", val: searchParam },
    { key: "keyword", val: search },
    { key: "pageSize", val: pageSize },
    { key: "pageNumber", val: pageNumber },
    { key: "paginate", val: paginate },
  ]);
  return axios.get(`${serverURL}/dorm${parsedParams}`);
};

export const fetchHomes = ({
  searchParam = "",
  search = "",
  pageSize = 10,
  pageNumber = 1,
  paginate = true,
}) => {
  const parsedParams = paramParser([
    { key: "param", val: searchParam },
    { key: "keyword", val: search },
    { key: "pageSize", val: pageSize },
    { key: "pageNumber", val: pageNumber },
    { key: "paginate", val: paginate },
  ]);
  return axios.get(`${serverURL}/house${parsedParams}`);
};

export const fetchSingleDormById = id => {
  const url = `${serverURL}/dorm/${id}`;
  return axios.get(url);
};

export const verifyToken = token => {
  // AXIOS INTERCEPTOR
  axios.interceptors.response.use(
    response => response,
    axiosResolvers["400-401"],
  );
  const url = `${serverURL}/users/auth`;
  const config = {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  };
  return axios.get(url, config);
};

export const signUpManagerRequest = data => {
  // AXIOS INTERCEPTOR
  axios.interceptors.response.use(
    response => response,
    axiosResolvers["400-401"],
  );

  const postData = {
    email: data.email,
    password: data.password,
    firstName: data.fName,
    lastName: data.fName,
    type: data.type,
  };
  return axios.post(`${serverURL}/users/manager`, postData);
};
