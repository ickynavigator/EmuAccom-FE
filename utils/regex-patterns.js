/* eslint-disable no-useless-escape */

const regexPatterns = {
  email: /^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,5})$/,
  name: /^[A-Za-z-]{1,64}$/,
  password: /^[A-Za-z0-9!@#$%^&*()_.]{4,64}$/,
};
export default regexPatterns;
