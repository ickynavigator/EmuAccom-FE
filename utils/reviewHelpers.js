/// <reference path="../types/typedefs.js" />

/**
 * Gets the star rating from all the ratings in the reviews array
 *
 * @param {Review[]} reviews - an array of reviews
 *
 */
export const getRating = reviews => {
  if (reviews.length <= 0) {
    return 0;
  }
  const rate = reviews.reduce((acc, review) => acc + review.rating, 0);
  return rate / reviews.length;
};

export default {};
