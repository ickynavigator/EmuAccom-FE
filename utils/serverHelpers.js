/* eslint-disable import/prefer-default-export */

export const isDev = () => {
  const environment = process.env.NODE_ENV;
  switch (environment) {
    case "production":
    case "staging":
      return false;
    case "development":
    case "testing":
    default:
      return true;
  }
};
