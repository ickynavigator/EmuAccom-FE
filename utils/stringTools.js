/* eslint-disable no-useless-escape */

export const regexPatterns = {
  text: /^[a-zA-Z0-9\s]*$/,
  email: /^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,5})$/,
  name: /^[A-Za-z-]{1,64}$/,
  password: /^[A-Za-z0-9!@#$%^&*()_.]{4,64}$/,
};

/**
 * Tests if a string is blank
 * @param {string} value
 * @returns {boolean}
 */
export function isBlank(value) {
  const regexp = /^ *$/;
  return regexp.test(value);
}

/**
 * tests if a string is a undefined/null or empty
 * @param {string | undefined | null} value
 * @returns {boolean}
 */
export function isBlankOrUndefined(value) {
  if (value === undefined || value == null) {
    return true;
  }
  return isBlank(value);
}
