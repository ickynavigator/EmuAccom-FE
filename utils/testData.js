// TODO add dummy data for this dorm and add it to the dorm response array
export const dummyDorm = {};

export const dormResponse = {
  dorms: [],
  pages: 10,
  page: 1,
};
