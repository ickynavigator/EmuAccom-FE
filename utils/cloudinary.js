import { Notifications as message } from "./Notifications";

/**
 *  Handles the cloudinary uploads and gets the url
 *
 *  @param {{
 *      fileToUpload: File,
 *      successCB: (secureUrl: string) => void,
 *      errorCB?: (err: any) => void,
 *      doneCB?: () => void
 *  }} props
 */
export async function handleFileUpload(props) {
  const { fileToUpload } = props;
  const formData = new FormData();

  formData.append("file", fileToUpload);
  formData.append("upload_preset", process.env.NEXT_PUBLIC_CLOUDINARY_PRESET);

  const options = { method: "POST", body: formData };

  fetch(process.env.NEXT_PUBLIC_CLOUDINARY_URL, options)
    .then(res => res.json())
    .then(res => {
      const { secure_url: secureUrl } = res;
      if (secureUrl) {
        message.success("Image Uploaded");

        if (props.successCB) props.successCB(secureUrl);
      } else {
        message.error("An error has occured. Please try again");
      }
    })
    .catch(err => {
      message.error("An error has occured. Please try again");

      if (props.errorCB) props.errorCB(err);
    })
    .finally(() => {
      if (props.doneCB) props.doneCB();
    });
}

/**
 * @param {File} preview
 */
export async function parseBlob(preview) {
  return fetch(preview)
    .then(r => r.blob())
    .then(
      blobFile =>
        new File([blobFile], preview.name || new Date().toDateString(), {
          type: preview.type,
          ...preview,
        }),
    )
    .then(res => res);
}
