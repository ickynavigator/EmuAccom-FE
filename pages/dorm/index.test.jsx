import { render } from "@testing-library/react";
import React from "react";
import { dormResponse } from "../../utils/testData";
import Index from "./index.page";

describe("Home", () => {
  it("matches the Dorm page Snapshot", () => {
    const { container } = render(<Index data={dormResponse} />);
    expect(container).toMatchSnapshot();
  });
});
