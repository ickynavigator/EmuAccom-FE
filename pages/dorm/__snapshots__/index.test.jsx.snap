// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the Dorm page Snapshot 1`] = `
<div>
  <div>
    <div
      class="mantine-Container-root mantine-len9jq"
    >
      <form>
        <div
          class="mantine-Group-root mantine-q21ep2"
        >
          <div
            class="mantine-TextInput-root mantine-Group-child mantine-919xyt"
          >
            <label
              class="mantine-riils3 mantine-TextInput-label"
              for="mantine-6np6vi7ul"
              id="mantine-6np6vi7ul-label"
            >
              Search
            </label>
            <div
              class="mantine-TextInput-wrapper mantine-12sbrde"
            >
              <div
                class="mantine-1hseney mantine-TextInput-icon"
              >
                <svg
                  class="icon icon-tabler icon-tabler-search"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <circle
                    cx="10"
                    cy="10"
                    r="7"
                  />
                  <line
                    x1="21"
                    x2="15"
                    y1="21"
                    y2="15"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-TextInput-withIcon mantine-riw2j3"
                id="mantine-6np6vi7ul"
                placeholder="Search for a home"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="mantine-8od8ev"
    >
      <div
        class="mantine-Alert-root mantine-Alert-light mantine-tloqvs"
      >
        <div
          class="mantine-17do188 mantine-Alert-wrapper"
        >
          <div
            class="mantine-soroux mantine-Alert-icon"
          >
            <svg
              class="icon icon-tabler icon-tabler-alert-circle"
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <circle
                cx="12"
                cy="12"
                r="9"
              />
              <line
                x1="12"
                x2="12"
                y1="8"
                y2="12"
              />
              <line
                x1="12"
                x2="12.01"
                y1="16"
                y2="16"
              />
            </svg>
          </div>
          <div
            class="mantine-1eawhj0 mantine-Alert-body"
          >
            <div
              class="mantine-lbapkq mantine-Alert-title"
            >
              <span
                class="mantine-194kea0 mantine-Alert-label"
              >
                Bummer!
              </span>
            </div>
            <div
              class="mantine-kgnty3 mantine-Alert-message"
            >
              Sorry but no Dorm has been found.
              <div
                class="mantine-a9v14h"
              />
              Please try different parameters
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
